### DISCUSSION TITLE
[NOT RELATED TO NEXT.JS] Typescript, semicolon or not?

### DISCUSSION THREAD

**Discussion by devjiwonchoi** [vercel/next.js#51388] [Help] [answered]
### Summary

During contributing to Next.js, I've noticed that we're not using semicolons on Typescript. 
I know that this is a controversial topic just wanted to know what the community thinks.

TL;DR: Use semicolons in Typescript?

P.S. If there are any official docs related to this subject please feel free to share.

### Additional information

```js
Microsoft, the father of Typescript uses a semicolon on their own `Typescript` project.
```

**Comment by icyJoseph** [comment, marked as answer]
I think this is often controlled by with Prettier. By using the `semi: false` setting: https://prettier.io/docs/en/options.html#semicolons

Which is to say, that this kind of discussion, in 2023, ought to be left out to automation. Yes, at some point the team/project has to decide, but once that decision is made, as you save, the code is formatted, or the prettier/linter tool can warn you about the usage, or lack there of, or even fix it on command. 

I had also made some commits to Next.js and never even notice there was no semi-colons 😅 - and I do, manually write them, but the linter/prettier fixed it up.

There's some cases where semi-colons convey the correct meaning. At the end of the day this is JS thing, more than it is a TypeScript. C# uses semi-colons. Rust does too. And the automation tools handle the semicolon, or lack thereof, with proper correctness. 

I bring up C# because [Anders Hejlsberg](https://en.wikipedia.org/wiki/Anders_Hejlsberg) has been the lead developer of C# and TypeScript. I guess it makes sense they just follow the C# syntax into their TypeScript work.

Last but not least, https://brendaneich.com/2012/04/the-infernal-semicolon/, a word from the Brendan Eich, this article has links to many spicy comments, for example https://github.com/twbs/bootstrap/issues/3057#issuecomment-5135512

**Reply by devjiwonchoi** [reply]
Cool, I absolutely agree with you. Could you share your reason why you set the config as no semicolon?

**Reply by icyJoseph** [reply]
Me? I think that was done back in 2019 or so 😅 that's up to the next team coding style. It's near bikeshedding territory. And somewhat controversial too. 

Semicolons are not required, but they are added, transparently to you, when the code is parsed. If you want to go without them, you better be sure about where they'll be automatically added.

**Reply by devjiwonchoi** [reply]
@icyJoseph Thank you for your helpful answer. Really meant a lot.

**Comment by devjiwonchoi** [comment]
Open to other opinions, won't close/mark as answered for now 😄
